### DISCUSSION TITLE
Handling Devices with multiple battery levels (Wyze Lock YD.LO1)

### DISCUSSION THREAD

**Discussion by DimWittedOne** [andrew-codechimp/HA-Battery-Notes#793] [Q&A]
Hi!

I have the original [Wyze Lock (YD.LO1)](https://www.wyze.com/products/wyze-lock?variant=42302751080610) that optionally uses a standalone keypad with its own batteries. Any suggestions how to handle multiple battery levels associated with a single device? I've included a screenshot of how it looks at the moment.

![image](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/16409056/43628114-ec4e-4f09-ae16-c20dd7fc8c67)

The lock itself uses 4xAA while the keypad uses 2xAA. The current entry in the battery library (#243) indicates it uses a rechargeable battery, but my variant does not.  I'm using the Wyze integration found [here](https://github.com/SecKatie/ha-wyzeapi#readme).

```
attribution: Data provided by Wyze
state: locked
available: true
door_open: false
device_model: YD.LO1
mac: YD.LO1.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
lock_battery: 96
keypad_battery: 30
friendly_name: Front Door
```

Thanks!

**Comment by andrew-codechimp** [comment]
Multiple battery's aren't something I've come across before, it can't play nice with the device screen where the battery level is shown.

Not sure how I could handle this, but I'll give it some thought.

**Comment by fetzerch** [comment]
I'd also be interested in a solution/recommendation. I have a few smoke detectors (https://www.somfy.co.uk/products/2401368/smoke-alarm-io) that have 2 batteries. A 9V one for the sensor itself and a CR123 for communication with the hub.
